tracer/test/Datadog.Trace.Tests/SpanContextPropagatorTests.cs

```diff
@@ -3,34 +3,322 @@
+
+        public static IEnumerable<object[]> GetInvalidIds()
+        {
+            yield return new object[] { null };
+            yield return new object[] { string.Empty };
+            yield return new object[] { "0" };
+            yield return new object[] { "-1" };
+            yield return new object[] { "id" };
+        }
+
+        [Fact]
+        public void Inject_IHeadersCollection()
+        {
+            var context = new SpanContext(TraceId, SpanId, SamplingPriority, serviceName: null, Origin);
+            var headers = new Mock<IHeadersCollection>();
+
+            SpanContextPropagator.Instance.Inject(context, headers.Object);
+
+            VerifySetCalls(headers);
+        }
+
+        [Fact]
+        public void Inject_CarrierAndDelegate()
+        {
+            var context = new SpanContext(TraceId, SpanId, SamplingPriority, serviceName: null, Origin);
+
+            // using IHeadersCollection for convenience, but carrier could be any type
+            var headers = new Mock<IHeadersCollection>();
+
+            SpanContextPropagator.Instance.Inject(context, headers.Object, (carrier, name, value) => carrier.Set(name, value));
+
+            VerifySetCalls(headers);
+        }
+
+        [Fact]
+        public void Extract_IHeadersCollection()
+        {
+            // use `object` so Should() below works correctly,
+            // otherwise it picks up the IEnumerable<KeyValuePair<string, string>> overload
+            var headers = SetupMockHeadersCollection();
+            object result = SpanContextPropagator.Instance.Extract(headers.Object);
+
+            VerifyGetCalls(headers);
+
+            result.Should()
+                  .BeEquivalentTo(
+                       new
+                       {
+                           TraceId,
+                           SpanId,
+                           Origin,
+                           SamplingPriority
+                       });
+        }
+
+        [Fact]
+        public void Extract_CarrierAndDelegate()
+        {
+            // using IHeadersCollection for convenience, but carrier could be any type
+            var headers = SetupMockHeadersCollection();
+
+            // use `object` so Should() below works correctly,
+            // otherwise it picks up the IEnumerable<KeyValuePair<string, string>> overload
+            object result = SpanContextPropagator.Instance.Extract(headers.Object, (carrier, name) => carrier.GetValues(name));
+
+            VerifyGetCalls(headers);
+
+            result.Should()
+                  .BeEquivalentTo(
+                       new
+                       {
+                           TraceId,
+                           SpanId,
+                           Origin,
+                           SamplingPriority
+                       });
+        }
+
+        [Fact]
+        public void Extract_ReadOnlyDictionary()
+        {
+            // using IHeadersCollection for convenience, but carrier could be any type
```

Comment:
[nit] irrelevant comment

Reply:
Thanks. Fixed in c9093f728360a4bbd0f89e96245f14cc70ade931.